### ISSUE TITLE
On decoding msgpack to cmt, map->metric_static_set is always ON

### ISSUE THREAD

**Issue by edsiper** [calyptia/cmetrics#19]
Just found an issue when converting from msgpack to cmt, the metrics maps always have metric_static_set to on. 

Example:

expected output:

```
# HELP kubernetes_network_load Network load
# TYPE kubernetes_network_load counter
kubernetes_network_load{hostname="localhost",app="cmetrics"} 1 1624856104590
kubernetes_network_load{hostname="localhost",app="test"} 12.15 1624856104590
```

real output:

```
# HELP kubernetes_network_load Network load
# TYPE kubernetes_network_load counter
kubernetes_network_load 0 0
# HELP kubernetes_network_load Network load
# TYPE kubernetes_network_load counter
kubernetes_network_load{hostname="localhost",app="cmetrics"} 1 1624856104590
kubernetes_network_load{hostname="localhost",app="test"} 12.15 1624856104590
```

it always includes the metric_static_set (but it was never set) by the caller:

```
kubernetes_network_load 0 0
```

this can be reproduced by using the following test snippets:

```C
#include <cmetrics/cmetrics.h>
#include <cmetrics/cmt_counter.h>
#include <cmetrics/cmt_encode_msgpack.h>
#include <cmetrics/cmt_decode_msgpack.h>
#include <cmetrics/cmt_encode_prometheus.h>
#include <cmetrics/cmt_encode_text.h>

#include "cmt_tests.h"

static struct cmt *generate_encoder_test_data()
{
    int ret;
    double val;
    uint64_t ts;
    struct cmt *cmt;
    struct cmt_counter *c;

    cmt = cmt_create();

    c = cmt_counter_create(cmt, "kubernetes", "network", "load", "Network load",
                           2, (char *[]) {"hostname", "app"});

    ts = cmt_time_now();

    ret = cmt_counter_inc(c, ts, 2, (char *[]) {"localhost", "cmetrics"});
    ret = cmt_counter_get_val(c, 2, (char *[]) {"localhost", "cmetrics"}, &val);
    ret = cmt_counter_add(c, ts, 10.55, 2, (char *[]) {"localhost", "test"});
    ret = cmt_counter_get_val(c, 2, (char *[]) {"localhost", "test"}, &val);
    ret = cmt_counter_set(c, ts, 12.15, 2, (char *[]) {"localhost", "test"});
    ret = cmt_counter_set(c, ts, 1, 2, (char *[]) {"localhost", "test"});

    return cmt;
}

/*
 * perform the following data encoding and compare msgpack buffsers
 *
 * CMT -> MSGPACK -> CMT -> MSGPACK
 *          |                  |
 *          |---> compare <----|
 */

void test_cmt_to_msgpack()
{
    int ret;
    char *mp1_buf = NULL;
    size_t mp1_size = 1;
    char *mp2_buf = NULL;
    size_t mp2_size = 2;
    struct cmt *cmt1 = NULL;
    struct cmt *cmt2 = NULL;

    /* Generate context with data */
    cmt1 = generate_encoder_test_data();
    TEST_CHECK(cmt1 != NULL);

    printf("%s\n", cmt_encode_prometheus_create(cmt1, CMT_TRUE));
    printf("--------\n");

    /* CMT1 -> Msgpack */
    ret = cmt_encode_msgpack(cmt1, &mp1_buf, &mp1_size);
    TEST_CHECK(ret == 0);

    /* Msgpack -> CMT2 */
    ret = cmt_decode_msgpack(&cmt2, mp1_buf, mp1_size);
    TEST_CHECK(ret == 0);

    /* CMT2 -> Msgpack */
    ret = cmt_encode_msgpack(cmt2, &mp2_buf, &mp2_size);
    TEST_CHECK(ret == 0);

    printf("%s\n", cmt_encode_prometheus_create(cmt2, CMT_TRUE));
    printf("--------\n");

    cmt_destroy(cmt1);
    cmt_destroy(cmt2);
    free(mp1_buf);
    free(mp2_buf);
}

TEST_LIST = {
    {"cmt_msgpack", test_cmt_to_msgpack},
    { 0 }
};
```

that's just a modified version of the encoding.c test, the difference is that metrics "without labels" are never set.

**Comment by edsiper** [comment]
maybe this ?

https://github.com/calyptia/cmetrics/blob/master/src/cmt_decode_msgpack.c#L354

**Comment by edsiper** [comment]
fixed in 683431ed
